### PR TITLE
[Undertow] refactor plain serializers

### PR DESCRIPTION
**Pull request by carterkozak**

### CODE REVIEW

File: conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java [hunk 1]
```diff
@@ -697,7 +697,7 @@ public void handleRequest(HttpServerExchange exchange) throws IOException {
             AuthHeader authHeader = runtime.auth().header(exchange);
             Map<String, Deque<String>> queryParams = exchange.getQueryParameters();
             String queryParamNameRaw =
-                    runtime.serde().deserializeString(queryParams.get("queryParamName"));
+                    runtime.serde().plain().deserializeString(queryParams.get("queryParamName"));
```
Comment (carterkozak) [inline conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java]: I think `runtime.plain()` is a cleaner here

Comment (carterkozak) [reply conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java]: @rubenfiszel thoughts?

Comment (carterkozak) [reply conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java]: Perhaps it works better if I also rename `SerDe` to `BodySerDe` and `UndertowRuntime.body()`?

File: conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java [hunk 2]
```diff
@@ -17,13 +17,19 @@
...
+ *
+ * The {@link UndertowRuntime} and provided interfaces {@link BodySerDe}, {@link Plain}, and
+ * {@link AuthorizationExtractor} are internal API, no guarantees are made for custom implementations.
  */
 public interface UndertowRuntime {
 
-    /** Provides the configured {@link SerDe}. */
-    SerDe serde();
+    /** Provides the {@link BodySerDe} used to deserialize and serialize request and response bodies respectively. */
+    BodySerDe body();
```
Comment (markelliot) [inline conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java]: bodySerDe?

File: conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java [hunk 3]
```diff
@@ -17,13 +17,19 @@
...
+
+    /** Provides the {@link Plain} used to parse request path, query, and header parameters. */
+    Plain plain();
```
Comment (markelliot) [inline conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java]: plainSerDe?

File: conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Plain.java [hunk 4]
```diff
@@ -33,22 +29,14 @@
...
-
-    /** Reads an {@link InputStream} from the {@link HttpServerExchange} request body. */
-    InputStream deserializeInputStream(HttpServerExchange exchange);
+/**
+ * Provides functionality to parse supported types using the
+ * <a href="https://github.com/palantir/conjure/blob/master/docs/spec/wire.md#6-plain-format">Conjure PLAIN format</a>.
+ *
+ * These utilities are used to parse HTTP path, query, and header parameter values.
+ */
+public interface Plain {
```
Comment (markelliot) [inline conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Plain.java]: PlainSerDe?

Comment (carterkozak) [reply conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Plain.java]: There's no `Ser`, only `De` for plain fields on servers, but if we did add serialization we would put it here. 👍

File: conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java (hunk 2 above)
Comment (carterkozak) [reply conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java]: updated

File: conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java (hunk 1 above)
Comment (rubenfiszel) [reply conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java]: Yes, either one or the other but SerDe and PlainSerDe don't make sense to coexist.

Comment (carterkozak) [reply conjure-java-core/src/integrationInput/java/com/palantir/product/EteServiceEndpoints.java]: Yep, updated to `PlainSerDe` and `BodySerDe`

**Pull request merged**